src/main/java/com/aws/greengrass/logmanager/model/LogFile.java

```diff
@@ -25,34 +25,42 @@ public class LogFile extends File {
     private static final Logger logger = LogManager.getLogger(LogManagerService.class);
     public static final int bytesNeeded = 1024;
     public static final String HASH_VALUE_OF_EMPTY_STRING = "";
+    @SuppressWarnings("PMD.UnusedPrivateField")
+    private final String sourcePath;
 
-    public LogFile(String pathname) {
-        super(pathname);
+    public LogFile(Path sourcePath, Path hardLinkPath) {
+        super(hardLinkPath.toString());
+        this.sourcePath = sourcePath.toString();
     }
 
     public LogFile(URI uri) {
         super(uri);
+        this.sourcePath = uri.getPath();
     }
 
     /**
      * Convert the file to LogFile.
      * @param file The file to be converted.
-     * @return
      */
     public static LogFile of(File file) {
-        return new LogFile(file.getAbsolutePath());
+        return new LogFile(file.toURI());
     }
 
     /**
-     * Convert list of files to list of LogFiles.
-     * @param files The list of files to be converted.
-     * @return
+     * Convert the file to LogFile.
+     *
+     * @param sourceFile        The file to be converted.
+     * @param hardLinkDirectory path where the hardlink should be stored
+     * @throws IOException if can't find the source path
      */
-    public static LogFile[] of(File... files) {
-        if (files == null) {
-            return new LogFile[] {};
-        }
-        return Arrays.stream(files).map(LogFile::of).toArray(LogFile[]::new);
+    public static LogFile of(File sourceFile, Path hardLinkDirectory) throws IOException {
+        // Why do we need this? - Hardlinks can't get the path of their source file. We need to keep track of which path
+        // it was originally created with and in case the file on that path changes we need to scan the directory to
+        // check for the inode that matches the hardlink to get the correct path.
+        Path destinationPath = hardLinkDirectory.resolve(sourceFile.getName());
```

Comment:
In case someone asks why have this here. It is because we will need to add a method to check if the file it was originally tracking is not the same one as the one the hardlink is tracking